### ISSUE TITLE
quote termination confused by '#' char and interpolated vars not recognized

### ISSUE THREAD

**Issue by mhshph** [bash-lsp/bash-language-server#723]
### Code editor

vscode

### Platform

OS X and linux

### Version

1.33.0

### What steps will reproduce the bug?

The following fragment demonstrate the problems:

```
# interpolated vars not recognized
foo="xx${bar}xx"

# quoted '#' char breaks termination of quote context:
a=( '#oops' )
echo this should not be in quote context
# '
# the above quote in comment ends the single quote context

a=( "#oops" )
echo problem does not occur with double quotes

# moving the leading quote also breaks quote termination:
a=( --foo="${bar}" )
echo again quote context not terminated 
#"
echo now back to normal

```

### How often does it reproduce? Is there a required condition?

reproduces reliably 

### What is the expected behavior?

Vars in double quote context should get distinct color.

A '#' char in sinlge quotes should not prevent the  end of quote from being recognized.

With:
`a=( --foo="${bar}" )
`
the quote context should stop at the second double quote. 

### What do you see instead?

![quoting-problem-macos](https://user-images.githubusercontent.com/93323705/217669757-abab2678-1639-476a-889f-f41b2497c6f9.png)
![quoting-problem-linux](https://user-images.githubusercontent.com/93323705/217669764-26638478-3497-4035-b38b-a08f6fa673ef.png)

### Additional information

Where can I find the language server logs?  I will gladly post them if I can find.

**Comment by mhshph** [comment]
Apparently the above is from the vscode default syntax highlighting, and not from the extension. On startup, there is a brief moment where the interpolated var is colored, and then the buffer reverts to the state above.   But when I remove the extension, I get the above same output. So maybe the extension is just crashing. If I cannot find log output shortly, then I will just close this.

**Comment by mhshph** [comment]
After enabling the debug setting for the extension, I have log output (below)
And I also confirmed that the extension is running ok and recognizes bash in the buffer. When I edit the sample script given above in the repro section, I can see in the log that the extension is getting the events.  e.g. when I click on 'bar' I see in the log:
`01:26:11.332 DEBUG onDocumentHighlight 15:15 word="bar"`
 but seems like it is leaving the syntax highlighting to the default vscode.


```
01:21:31.774 DEBUG Initializing...
01:21:32.617 DEBUG Initialized
01:21:32.646 DEBUG Configuration loaded from client
[Info  - 9:21:32 AM] 01:21:32.695 INFO BackgroundAnalysis: resolving glob "wraps/*@(.sh|.inc|.bash|.command)" inside "file:///sw/vscode~pre"...
01:21:32.788 DEBUG onCodeAction: found 0 code action(s)
[Info  - 9:21:32 AM] 01:21:32.805 INFO BackgroundAnalysis: Glob resolved with 2 files after 0.081 seconds
[Info  - 9:21:32 AM] 01:21:32.837 INFO BackgroundAnalysis: Completed after 0.113 seconds.
01:21:33.118 DEBUG onCodeAction: found 0 code action(s)
01:21:33.125 DEBUG onCodeAction: found 0 code action(s)
01:21:33.196 DEBUG ShellCheck: running "shellcheck --shell=bash --format=json1 --external-sources --source-path=file:///sw/vscode~pre --source-path=/sw/vscode~pre/wraps"
01:21:33.612 DEBUG onCodeAction: found 0 code action(s)
01:21:34.072 DEBUG onCodeAction: found 0 code action(s)
01:21:34.106 DEBUG onDocumentSymbol
01:21:34.107 DEBUG onDocumentSymbol
```

**Comment by mhshph** [comment]
After further investigation it appears that Bash IDE does not do the syntax highlighting, in which case this is a vscode issue. I will submit an issue there.
